src/components/home/vacations-card.tsx

```diff
@@ -21,14 +29,18 @@ const VacationsCard = () => {
   const { vacationRequestsApi, vacationRequestStatusApi } = useApi();
   const userProfile = useAtomValue(userProfileAtom);
   const setError = useSetAtom(errorAtom);
-  const persons = useAtomValue(personsAtom);
   const [vacationRequests, setVacationRequests] = useAtom(vacationRequestsAtom);
+  const initialVacationRequestStatuses = useAtomValue(vacationRequestStatusesAtom);
   const [vacationRequestStatuses, setVacationRequestStatuses] = useState<VacationRequestStatus[]>(
-    []
+    initialVacationRequestStatuses
   );
   const setLatestVacationRequestStatuses = useSetAtom(vacationRequestStatusesAtom);
   const language = useAtomValue(languageAtom);
   const [loading, setLoading] = useState(false);
+  const adminMode = UserRoleUtils.adminMode();
+  const persons = useAtomValue(personsAtom);
+  const location = useLocation();
+  const keepData = location.state?.keepData;
```

Comment:
If atom was used for this then it would need some more logic to keep track of the last route the user was in. useLocation just conveniently happens to clear the keepData state when the user navigates between the main routes:  "/" and "/admin". I'll change the state name to a more descriptive one if that's okay.

Reply:
Renamed to keepVacationsData

Reply:
I just noticed that the state does not persist when using the browser forwards/backwards buttons, maybe custom logic is needed after all.

Reply:
Resolved by separating vacationsAtom and allVacationsAtom for admins.